ui/Modal/ProjectCreation.svelte

```diff
@@ -28,131 +33,30 @@
   } from "../DesignSystem/Component";
 
   let currentSelection;
+  let nameInput;
   export let content;
 
-  const projectNameMatch = "^[a-z0-9][a-z0-9._-]+$";
   const dispatch = createEventDispatcher();
 
   $: isNew = currentSelection === RepoType.New;
   $: isExisting = currentSelection === RepoType.Existing;
 
   let name;
   let description = "";
-  let defaultBranch = DEFAULT_BRANCH_FOR_NEW_PROJECTS;
   let newRepositoryPath = "";
   let existingRepositoryPath = "";
 
-  let validations = false;
-  let beginValidation = false;
+  let nameValidation = nameValidationStore();
 
   let loading = false;
 
-  validatejs.options = {
-    fullMessages: false,
-  };
-
-  validatejs.validators.optional = (value, options) => {
-    return !validatejs.isEmpty(value)
-      ? validatejs.single(value, options)
-      : null;
-  };
-
-  validatejs.validators.validateNewRepositoryPath = (
-    value,
-    _options,
-    _key,
-    _attributes
-  ) => {
-    if (isExisting) {
-      return;
-    }
-
-    if (validatejs.isEmpty(value)) {
-      return "Pick a directory for the new project";
-    }
-
-    if (!localStateError.match("could not find repository")) {
-      return "The directory should be empty";
-    }
-  };
-
-  validatejs.validators.validateExistingRepositoryPath = (
-    value,
-    _options,
-    _key,
-    _attributes
-  ) => {
-    if (isNew) {
-      return;
-    }
-
-    if (validatejs.isEmpty(value)) {
-      return "Pick a directory with an existing repository";
-    }
-
-    if (localStateError.match("could not find repository")) {
-      return "The directory should contain a git repository";
-    }
-  };
-
-  const constraints = {
-    name: {
-      presence: {
-        message: "Project name is required",
-        allowEmpty: false,
-      },
-      format: {
-        pattern: new RegExp(projectNameMatch, "i"),
-        message: `Project name should match ${projectNameMatch}`,
-      },
-    },
-    currentSelection: {
-      presence: {
-        message:
-          "Select whether to start a new repository or use an existing one",
-      },
-    },
-    newRepositoryPath: {
-      validateNewRepositoryPath: true,
-    },
-    existingRepositoryPath: {
-      validateExistingRepositoryPath: true,
-    },
-  };
-
-  const validate = () => {
-    if (!beginValidation) {
-      return;
-    }
-
-    validations = validatejs(
-      {
-        name: name,
-        currentSelection: currentSelection,
-        newRepositoryPath: newRepositoryPath,
-        existingRepositoryPath: existingRepositoryPath,
-      },
-      constraints
-    );
+  const setSelection = type => {
```

Comment:
`setSelection` is a little vague; something like `setRepoType` would be more clear.

Reply:
Renamed it to `setCurrentSelection`